applications/infra.py

```diff
@@ -0,0 +1,241 @@
+# E.g. run with
+# export PYTHONPATH=${PYTHONPATH}:.
+# python applications/infra.py
+# or (macos):
+# pythonw applications/infra.py
+
+import matplotlib.pyplot as plt
+import nevergrad as ng
+import numpy as np
+import random
+import math
+pi = math.pi
+
+# We consider a power systems problem.
+# We have N stocks.
+# Each stock is managed by an agent.
+# Agents collaborate for minimizing the total cost.
+
+# Number of stocks (dams).
+N = 7
+
+# Optimization budget.
+optim_steps = 8
+
+# Real life is more complicated! This is a very simple model.
+
+class Agent():
+    """An agent has an input size, an output size, a number of layers, a width of its internal layers 
+    (a.k.a number of neurons per hidden layer)."""
+    def __init__(self, input_size, output_size, layers=2, layer_width=20):
+        assert(layers >= 2)
+        self.input_size = input_size
+        self.output_size = output_size
+        self.layers = []
+        self.layers += [np.random.rand(int(input_size), int(layer_width))]
+        for i in range(layers-2):
+            self.layers += [np.random.rand(int(layer_width), int(layer_width))]
+        self.layers += [np.random.rand(int(layer_width), int(output_size))]
+        assert len(self.layers) == layers
+
+    def GetParamNumbers(self):
+        return sum([np.prod(l.shape) for l in self.layers])
+
+    def SetParams(self, ww):
+        w = [w for w in ww]
+        assert(len(w) == self.GetParamNumbers())
+        for l in self.layers:
+            s = np.prod(l.shape)
+            l = np.reshape(np.array(w[:s]), l.shape)
+            w = w[s:]
+
+    def GetOutput(self, inp):
+        output = np.array(inp).reshape(1, len(inp))
+        for l in [self.layers[i] for i in range(len(self.layers) - 1)]:
+            output = np.tanh(np.matmul(output, l))
+        return np.matmul(output, self.layers[-1])
+         
+
+
+
+
+
+dam_managers = []
+
+# Parameters describing the problem.
+year_to_day_ratio = 1.  # Ratio between variation of consumption in the year and variation of consumption in the day
+constant_to_year_ratio = 1.
+dam_output_factor = 1.0
+back_to_normal = 0.5  # How much of the gap with normal is cancelled at each iteration.
+consumption_noise = 0.1
+num_thermal_plants = 7
+number_of_years = 1
+
+average_consumption = constant_to_year_ratio * year_to_day_ratio
+thermal_power_capacity = [c * average_consumption for c in list(np.random.rand(num_thermal_plants))]
```

Comment:
So that probably won't make much difference here because this is called only once but np.random.rand(num_thermal_plants) is already iterable so you don't need the list() thing. Multiplying by c would be more efficient and I think you can keep it as a numpy array:
` thermal_power_capacity = average_consumption * np.random.rand(num_thermal_plants)`
If you really want a list in the end you can write it `list(average_consumption * np.random.rand(num_thermal_plants))`

Reply:
I remove this file but report the corresponding modification in the other similar file.